### PR TITLE
Algorithm parameter support for DBSCAN

### PR DESCRIPTION
**Pull request by Er1cCheng**
Add in algorithm parameter support for DBSCAN, could choose between "brute" and "rbc" algorithm

### CODE REVIEW

**Comment by Er1cCheng** [comment]
build

**Comment by Er1cCheng** [comment]
build

File: python/tests/test_dbscan.py [hunk 1]
```diff
@@ -166,13 +164,17 @@ def test_dbscan_numeric_type(gpu_number: int, data_type: str) -> None:
     ids=idfn,
 )
 @pytest.mark.parametrize("data_type", cuml_supported_data_types)
-@pytest.mark.parametrize("max_record_batch", [100, 10000])
+@pytest.mark.parametrize(
+    "max_record_batch", [pytest.param(100, marks=pytest.mark.slow), 10000]
+)
+@pytest.mark.parametrize("algorithm", ["brute", "rbc"])
```
Comment (eordentlich) [inline python/tests/test_dbscan.py:170]: Is this test run time still reasonable?

Comment (Er1cCheng) [reply python/tests/test_dbscan.py:170]: Yes, it's actually pretty fast given it runs on 4 GPUs.

Comment (Er1cCheng) [reply python/tests/test_dbscan.py:170]: And that some parameters are marked as slow so not all of them are executed for CI.

**Review by lijinf2 — APPROVED** [review]
Looks good.

**Pull request merged**